### PR TITLE
executor,infoschema: move `dataForTableConstraints` into executor

### PR DESCRIPTION
**Pull request by Rustin-Liu**
UCP #15027 
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve? 
move `dataForTableConstraints` into executor.

### What is changed and how it works?
move `dataForTableConstraints` into executor.

### Check List 

Tests 

 - Unit test

Code changes

None
Side effects

None

Related changes

 None
Release note
None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@zz-jason  @lonng  PTAL~

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ import (
 	"github.com/pingcap/tidb/util/sqlexec"
 )
 
+const (
```
Comment (lonng) [inline executor/infoschema_reader.go]: It's better to keep them in `infoschema` package.

Comment (Rustin-Liu) [reply executor/infoschema_reader.go]: Fixed, PTAL~

**Comment by Rustin-Liu** [comment]
@zz-jason @lonng  ping~

**Pull request merged**